Move Half to headeronly

janeyx99 opened the pull request:
Essence of this copypasta:
- combine Half-inl.h and Half.h in c10/util -> torch/headeronly/util/Half.h
- Add NOLINTNEXTLINE's to the portions of Half-inl.h that were previously in the ignore list of clangtidy
- Re-expose all APIs in namespaces and through includes of the original files. Ideally, we would have the APIs in torch::headeronly and reexpose them in c10, but that runs into BC issues (see D78997465) so for now we are keeping the APIs in c10 but reexposing them in torch::headeronly.
- Change test cases in test_aoti_abi_check to test torch::headeronly::Half vs c10::Half (they're the same thing but we eventually want all the tests for headeronly APIs to only import from headeronly).

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #159172

janeyx99 committed:
Move Half to headeronly

[ghstack-poisoned]

janeyx99 committed:
Update on "Move Half to headeronly"




[ghstack-poisoned]

albanD reviewed (approved):
Nice!

janeyx99 committed:
Update on "Move Half to headeronly"

Essence of this copypasta:
- combine Half-inl.h and Half.h in c10/util -> torch/headeronly/util/Half.h
- Add NOLINTNEXTLINE's to the portions of Half-inl.h that were previously in the ignore list of clangtidy
- Re-expose all APIs in namespaces and through includes of the original files.
- Change test cases in test_aoti_abi_check to test torch::headeronly::Half vs c10::Half (they're the same thing but we eventually want all the tests for headeronly APIs to only import from headeronly).




[ghstack-poisoned]

janeyx99 committed:
Update on "Move Half to headeronly"

Essence of this copypasta:
- combine Half-inl.h and Half.h in c10/util -> torch/headeronly/util/Half.h
- Add NOLINTNEXTLINE's to the portions of Half-inl.h that were previously in the ignore list of clangtidy
- Re-expose all APIs in namespaces and through includes of the original files.
- Change test cases in test_aoti_abi_check to test torch::headeronly::Half vs c10::Half (they're the same thing but we eventually want all the tests for headeronly APIs to only import from headeronly).




[ghstack-poisoned]

desertfire commented on torch/headeronly/util/Half.h:
```diff
@@ -241,9 +297,491 @@ inline uint16_t fp16_ieee_from_fp32_value(float f) {
+
+namespace c10 {
+
+using torch::headeronly::Half;
```
This reminded me some BC problems I ran into due to Argument-Dependent Lookup. Probably fine for this PR, but I feel like you may run into that again at some point.

janeyx99 replied:
Nah--you are right. I ran into them internally too (used another non ghexport PR to check D78997465). I'm gonna keep the c10: and re-expose c10 in torch::headeronly for now to avoid BC.

janeyx99 committed:
Update on "Move Half to headeronly"

Essence of this copypasta:
- combine Half-inl.h and Half.h in c10/util -> torch/headeronly/util/Half.h
- Add NOLINTNEXTLINE's to the portions of Half-inl.h that were previously in the ignore list of clangtidy
- Re-expose all APIs in namespaces and through includes of the original files.
- Change test cases in test_aoti_abi_check to test torch::headeronly::Half vs c10::Half (they're the same thing but we eventually want all the tests for headeronly APIs to only import from headeronly).




[ghstack-poisoned]

janeyx99 committed:
Update on "Move Half to headeronly"

Essence of this copypasta:
- combine Half-inl.h and Half.h in c10/util -> torch/headeronly/util/Half.h
- Add NOLINTNEXTLINE's to the portions of Half-inl.h that were previously in the ignore list of clangtidy
- Re-expose all APIs in namespaces and through includes of the original files. Ideally, we would have the APIs in torch::headeronly and reexpose them in c10, but that runs into BC issues (see D78997465) so for now we are keeping the APIs in c10 but reexposing them in torch::headeronly.
- Change test cases in test_aoti_abi_check to test torch::headeronly::Half vs c10::Half (they're the same thing but we eventually want all the tests for headeronly APIs to only import from headeronly).




[ghstack-poisoned]

janeyx99 committed:
Update on "Move Half to headeronly"

Essence of this copypasta:
- combine Half-inl.h and Half.h in c10/util -> torch/headeronly/util/Half.h
- Add NOLINTNEXTLINE's to the portions of Half-inl.h that were previously in the ignore list of clangtidy
- Re-expose all APIs in namespaces and through includes of the original files. Ideally, we would have the APIs in torch::headeronly and reexpose them in c10, but that runs into BC issues (see D78997465) so for now we are keeping the APIs in c10 but reexposing them in torch::headeronly.
- Change test cases in test_aoti_abi_check to test torch::headeronly::Half vs c10::Half (they're the same thing but we eventually want all the tests for headeronly APIs to only import from headeronly).




[ghstack-poisoned]

janeyx99 commented:
@pytorchbot merge

janeyx99 committed:
Update on "Move Half to headeronly"

Essence of this copypasta:
- combine Half-inl.h and Half.h in c10/util -> torch/headeronly/util/Half.h
- Add NOLINTNEXTLINE's to the portions of Half-inl.h that were previously in the ignore list of clangtidy
- Re-expose all APIs in namespaces and through includes of the original files. Ideally, we would have the APIs in torch::headeronly and reexpose them in c10, but that runs into BC issues (see D78997465) so for now we are keeping the APIs in c10 but reexposing them in torch::headeronly.
- Change test cases in test_aoti_abi_check to test torch::headeronly::Half vs c10::Half (they're the same thing but we eventually want all the tests for headeronly APIs to only import from headeronly).




[ghstack-poisoned]

janeyx99 commented:
@pytorchbot merge -f "Rebased, all tests passed before that were relevant"

clee2000 commented:
@pytorchbot revert -m "broke lint [GH job link](https://github.com/pytorch/pytorch/actions/runs/16613893793/job/47002486679) [HUD commit link](https://hud.pytorch.org/pytorch/pytorch/commit/6d0f4566e2b6e05369d8bb6c0d0e83a0eee982aa).  Note to self: why isn't Dr. CI updating" -c ignoredsignal

janeyx99 committed:
Update on "Move Half to headeronly"

Essence of this copypasta:
- combine Half-inl.h and Half.h in c10/util -> torch/headeronly/util/Half.h
- Add NOLINTNEXTLINE's to the portions of Half-inl.h that were previously in the ignore list of clangtidy
- Re-expose all APIs in namespaces and through includes of the original files. Ideally, we would have the APIs in torch::headeronly and reexpose them in c10, but that runs into BC issues (see D78997465) so for now we are keeping the APIs in c10 but reexposing them in torch::headeronly.
- Change test cases in test_aoti_abi_check to test torch::headeronly::Half vs c10::Half (they're the same thing but we eventually want all the tests for headeronly APIs to only import from headeronly).




[ghstack-poisoned]

janeyx99 commented:
@pytorchbot merge -f "Lint is all green now, sorry for the landrace earlier"

yangw-dev commented:
@pytorchmergebot  revert -m "broken the  Lint for test/cpp/aoti_abi_check/test_dtype.cpp:" -c nosignal

https://github.com/pytorch/pytorch/actions/runs/16613893793/job/47002486679
